src/main/webapp/js/richTextEditor.js

```diff
@@ -1,6 +1,49 @@
 /* eslint-disable camelcase */ // The property names are determined by external library (tinymce)
-var richTextEditorBuilder = {
-    getDefaultConfiguration: function() {
+function RichTextEditor(params) {
+
+    // ******************************************************************************************************
+    // Private properties
+    // ******************************************************************************************************
+
+    var MANDATORY_INIT_PARAMS = ['selector'];
+    var initParams = {};
+
+    var MANDATORY_ADDITIONAL_PARAMS = ['uploadImageId', 'uploadImageInputName', 'createImageUploadUrl'];
```

Comment:
These "additional" params look like they are specifically targeted for image uploads. Perhaps a better name is "image upload" params?